File: iOS/Freelancer/Freelancer/SwiftUI_Backend/Data/Job/Job.swift
//
//  Job.swift
//  Freelancer
//
//  Created by Kiss Dániel Márk on 2021. 11. 09..
//
//

import Foundation

class Job: Transferable
{
    var link: String = "http://freelancerbackend-env.eba-34kjxuhr.eu-central-1.elasticbeanstalk.com/api/v1/jobs/"
    let jobHandler = JobDataHandler()
    var data: JobData
    var item: Item
    
    init(inData: JobData, itemIn : Item)
    {
        print("Job init....")
        data = inData
        item = itemIn
    }
    
    func createJSON() -> [String : Any] {
        let parameters: [String: Any] = [
            "destination": item.data.destination,
            "id" : item.data.id,
            "properties": item.data.propertis,
            "source":[
                "id": item.data.source.id,
                "location": item.data.source.location,
                "owner":[
                    "username":item.data.source.ownerName]
            ],
            "status":"TO_BE_DELIVERED"
        ]
        
        return parameters
    }
    
}

File: frontend/freelancer/src/app/entities/job.ts
import { SourceDTO } from "./source";
import { User } from "./user";

export class JobDto {
    id?:number;
    item?:Item;
    freelancer?:User;
}

export class Item {
    id?:number
    properties?:string
    destiantion?:string
    status?:string
    source?:SourceDTO
}

export class Job{
    id?:number;
    itemLocation?:string;
    itemDestination?:string;
    itemStatus?:string;
    freelancer?:string;

    constructor(jobDto:JobDto){
        this.id = jobDto.id
        this.itemDestination = jobDto.item?.destiantion
        this.itemLocation = jobDto.item?.source?.location
        this.itemStatus = jobDto.item?.status
        this.freelancer = jobDto.freelancer?.username
    }
}